src/octoprint/static/js/app/viewmodels/temperature.js

```diff
@@ -200,13 +200,40 @@ $(function () {
                 .heatedChamber.subscribe(self._printerProfileUpdated);
         });
 
+        self.MARKING_TYPE = {
+            PRINT: "print",
+            PAUSE: "pause",
+            RESUME: "resume",
+            CANCEL: "cancel",
+            DONE: "done",
+        };
+
+        self.markings = [];
+
+        self.marking_props = {};
+        self.marking_props[self.MARKING_TYPE.PRINT] = gettext("Start");
+        self.marking_props[self.MARKING_TYPE.PAUSE] = gettext("Pause");
+        self.marking_props[self.MARKING_TYPE.RESUME] = gettext("Resume");
+        self.marking_props[self.MARKING_TYPE.CANCEL] = gettext("Cancel");
+        self.marking_props[self.MARKING_TYPE.CANCEL] = gettext("Cancel");
+        self.marking_props[self.MARKING_TYPE.DONE] = gettext("Done");
+
+
+        self.markingColors = {};
+        self.markingColors[self.MARKING_TYPE.PRINT] = "#218656";
+        self.markingColors[self.MARKING_TYPE.PAUSE] = "#FDC02F";
+        self.markingColors[self.MARKING_TYPE.RESUME] = "#27CAEE";
+        self.markingColors[self.MARKING_TYPE.CANCEL] = "#DA3749";
+        self.markingColors[self.MARKING_TYPE.DONE] = "#1B72F9";
+
```

Comment:
```suggestion
        self.marking_props = {
            "print": {
                "label": gettext("Start"),
                "color": "#218656"
            },
            "pause": {
                "label": gettext("Pause"),
                "color": "#FDC02F"
            },
            "resume": {
                "label": gettext("Resume"),
                "color": "#27CAEE"
            },
            "cancel": {
                "label": gettext("Cancel"),
                "color": "#DA3749"
            },
            "done": {
                "label": gettext("Done"),
                "color": "#1B72F9"
            },
        }
```

And then use with `self.marking_props[mark.type].label` or `self.marking_props[mark.type].color`. Way less repetition and only one props object needed instead of multiple.